src/infi/clickhouse_orm/system_models.py

```diff
@@ -0,0 +1,142 @@
+"""
+from .database import Database  # Can't import it globally, due to circular import
+from .fields import *
+from .models import Model
+from .engines import MergeTree
+
+
+class SystemPart(Model):
+    """
+    Contains information about parts of a table in the MergeTree family.
+    This model operates only fields, described in the reference. Other fields are ignored.
+    https://clickhouse.yandex/reference_en.html#system.parts
+    """
+    OPERATIONS = frozenset({'DETACH', 'DROP', 'ATTACH', 'FREEZE', 'FETCH'})
+
+    readonly = True
+
+    database = StringField()  # Name of the database where the table that this part belongs to is located.
+    table = StringField()  # Name of the table that this part belongs to.
+    engine = StringField()  # Name of the table engine, without parameters.
+    partition = StringField()  # Name of the partition, in the format YYYYMM.
+    name = StringField()  # Name of the part.
+    replicated = UInt8Field()  # Whether the part belongs to replicated data.
+
+    # Whether the part is used in a table, or is no longer needed and will be deleted soon.
+    # Inactive parts remain after merging.
+    active = UInt8Field()
+
+    # Number of marks - multiply by the index granularity (usually 8192)
+    # to get the approximate number of rows in the part.
+    marks = UInt64Field()
+
+    bytes = UInt64Field()  # Number of bytes when compressed.
+
+    # Time the directory with the part was modified. Usually corresponds to the part's creation time.
+    modification_time = DateTimeField()
+    remove_time = DateTimeField()  # For inactive parts only - the time when the part became inactive.
+
+    # The number of places where the part is used. A value greater than 2 indicates
+    # that this part participates in queries or merges.
+    refcount = UInt32Field()
+
+    @classmethod
+    def table_name(cls):
+        return 'system.parts'
+
+    """
+    Next methods return SQL for some operations, which can be done with partitions
+    https://clickhouse.yandex/reference_en.html#Manipulations with partitions and parts
+    """
+    def _partition_operation_sql(self, db, operation, settings=None, from_part=None):
+        """
+        Performs some operation over partition
+        :param db: Database object to execute operation on
+        :param operation: Operation to execute from SystemPart.OPERATIONS set
+        :param settings: Settings for executing request to ClickHouse over db.raw() method
+        :return: Operation execution result
+        """
+        operation = operation.upper()
+        assert operation in self.OPERATIONS, "operation must be in [%s]" % ', '.join(self.OPERATIONS)
+        sql = "ALTER TABLE `%s`.`%s` %s PARTITION '%s'" % (db.db_name, self.table, operation, self.partition)
+        if from_part is not None:
+            sql += " FROM %s" % from_part
+        db.raw(sql, settings=settings, stream=False)
+
+    def detach(self, database, settings=None):

```

Comment:
SystemPart model is readonly. It is not saved in the database - it is system generated.
The only goal, I use the database - is filtering parts by it's name. By default system.parts table returns data for all databases in the system - it's starnge, but it is so.
1. I though of making database methods for working with partitions, alike insert and select, and they can be written on SystemPart bases. But to my mind this duplicating looks crazy.
2. I though of making setting database in __init__ method, but it's not usable for static get() and get_active() methods.
3. Setting database for each instance in get() method makes it incompatible with standard select - If someone makes database.select('SELECT * FROM system.parts', model_class=SystemPart) It will not save the database and operations will not be available.